### PR TITLE
Collaborative renaming & moving of files

### LINKED ISSUE #10327: Shared editing session is lost when renaming a file
## Description

With collaborative editing enabled (`--collaborative`): When renaming a file, the shared editing session is lost and new clients won't be able to collaborate. 

This issue was first reported in our working repository and is a known bug: https://github.com/jupyter-rtc/jupyterlab/issues/34

## Reproduce

1. Create a new notebook with collaborative editing enabled. E.g. using a binder instance: https://gist.github.com/jtpio/6ce26381703355e0ef1da4af742b7f72
2. Rename the file
3. Open another client to work on the same file
4. ⇒ The clients work on different editing sessions and do not sync content. 
5. ⇒ There is no loss of content because the clients will receive a "File Changed" dialogue when saving a file.

## Expected behavior

The second client will connect to the same shared-editing session as the first client.

## Context

- JupyterLab version: 3.1.0a10

### PR DESCRIPTION
**Pull request by dmonad**
## References

This PR fixes #10327 by updating the session context of remote peers. 

## Code changes

A collaborative session is still identified by the path. Now we can rename that path and notify other peers that they should use a new path.

## User-facing changes

Users can now rename files and folders while in a collaborative session. Remote users will automatically update the connection url and stay in the same collaborative session.

### CODE REVIEW

File: jupyterlab/handlers/yjs_echo_ws.py [hunk 1]
```diff
@@ -13,6 +13,7 @@
 releaseLockMessageType = 126
 requestInitializedContentMessageType = 125
 putInitializedContentMessageType = 124
+renameSession = 123
```
Comment (fcollonval) [inline jupyterlab/handlers/yjs_echo_ws.py:16]: I would suggest renaming it for consistency:

```suggestion
renameSessionMessageType = 123
```

Is there a reason why those message type are not defined as an [`IntEnum`](https://docs.python.org/3/library/enum.html#enum.IntEnum)?

Comment (dmonad) [reply jupyterlab/handlers/yjs_echo_ws.py]: Thanks! I implemented your suggestion.

I switched to IntEnum. The only reason why I didn't use this before was that I didn't know about it ¯\_(ツ)_/¯ (I'm a Python newbie).

**Comment by dmonad** [comment]
I rebased onto the current master. This should hopefully fix the broken tests.

**Review by blink1073 — APPROVED** [review]
Thanks!

File: jupyterlab/handlers/yjs_echo_ws.py (hunk 1 above)
Comment (fcollonval) [reply jupyterlab/handlers/yjs_echo_ws.py]: Thanks for making the changes @dmonad

**Pull request merged**